src/box/sql/expr.c

```diff
@@ -2801,7 +2801,8 @@ sqlCodeSubselect(Parse * pParse,	/* Parsing context */
 							    &sqlIntTokens[1]);
 				ExprSetProperty(pSel->pLimit, EP_System);
 			}
-			pSel->selFlags |= SF_SingleRow;
+			if (pExpr->op == TK_SELECT)
+				pSel->selFlags |= SF_SingleRow;
```

Comment:
In the comments above I see `the query is augmented with "LIMIT 1".  Any preexisting limit is discarded in place of the new LIMIT 1.` Doesn't it mean the code internally should only return 0 or 1 rows? If it fails on > 1 rows, it looks like the limit got lost somewhere and we actually perform the whole query even when <= 1 rows was enough.

Reply:
Thank you for the review! And sorry for such a long answer.

I believe this comment is untrue since https://github.com/tarantool/tarantool/commit/2a720d114dbd604a4f40a1d7b659a4abece2e646. In this commit, the implicit `LIMIT 1` was replaced with an error if more than one row was returned. I believe the fix was for subqueries, and `EXISTS` was changed along the way. I changed EXISTS to work the old way and corrected the comment accordingly. Also, I dropped doc-request since this is just a bug.